tests/integration-tests/src/test/resources/reload-address-queues-updated.xml

```diff
@@ -111,7 +111,12 @@ under the License.
             <config-delete-addresses>FORCE</config-delete-addresses>
          </address-setting>
       </address-settings>
-      
+
+      <wildcard-addresses>
+         <enabled>true</enabled>
```

Comment:
FWIW, the matching logic will use the configured <delimiter>, <any-words>, and <single-word> regardless of how <enabled> is set.  However, the routing logic won't.  Not sure if we want to make those semantics match (no pun intended).

Reply:
What do you do if false? Don't match? As that's what filtering does, it disables the ability to filter, it doesn't set the values to a default.

Reply:
The original flag was better name routing-enabled, maybe the config option should be renamed? (With still supporting enabled, as not major release, so can't break bits)

Reply:
I believe the only place isEnabled is checked is when the PostOffice is constructed.  If it is enabled then a `org.apache.activemq.artemis.core.postoffice.impl.WildcardAddressManager` is used otherwise a `org.apache.activemq.artemis.core.postoffice.impl.SimpleAddressManager` is used.  This effectively enables/disables wildcard address *routing*.  This is technically different than address *matching*.

Reply:
I agree...the original parameter name was better.  The fact that it was changed is probably also an indication as to why the matching logic was left untouched.

Reply:
question is do we want to rename the config to "routing-enabled" to make this clearer? Or just we need to make it clear in doc. The old config is still there so maybe the other option is we actually think of deprecating the "newer" enabled and un deprecate the older better named one

Reply:
My personal would be to think about the last option.

Reply:
I think your first option is the best.  My overall order preference would be 1, 2, 3.

Reply:
have updated pr with option 1.